crates/core/src/fs/addon.rs

```diff
@@ -19,6 +20,35 @@ pub fn delete_addons(addon_folders: &[AddonFolder]) -> Result<()> {
     Ok(())
 }
 
+/// Deletes all saved varaible files correlating to `[AddonFolder]`.
+pub fn delete_saved_variables(addon_folders: &[AddonFolder], wtf_path: &PathBuf) -> Result<()> {
+    for entry in WalkDir::new(&wtf_path)
+        .into_iter()
+        .filter_map(std::result::Result::ok)
+    {
+        let path = entry.path();
+        let file_name = path
+            .parent()
+            .and_then(|a| a.file_name())
+            .and_then(|a| a.to_str());
+
+        if file_name == Some("SavedVariables") {
+            let file_name = path.file_name().and_then(|f| f.to_str());
+
+            // NOTE: Will reject "Foobar_<invalid utf8>".
+            if let Some(file_name_str) = file_name {
+                for folder in addon_folders {
+                    if file_name_str.contains(&folder.id) {
```

Comment:
i haven't researched, but shouldn't the .lua file be named identically to the addon folder? The contains seems dangerous as it might remove things from similarly named addons that weren't intended to be deleted.

Reply:
You are right. Will fix this.